process.c

```diff
@@ -8779,16 +8779,58 @@ proc_warmup(VALUE _)
- *
- * - Process.exec: Replaces the current process by running a given external command.
- * - Process.spawn, Kernel#spawn: Executes the given command and returns its pid without waiting for completion.
- * - Kernel#system: Executes the given command in a subshell.
- *
- * Each of these methods accepts:
- *
- * - An optional hash of environment variable names and values.
- * - An optional hash of execution options.
+ * Each of the following methods executes a given command in a new process or subshell,
+ * or multiple commands in new processes and/or subshells.
+ * The choice of process or subshell depends on the form of the command;
+ * see {Argument command_line or exe_path}[rdoc-ref:Process@Argument+command_line+or+exe_path].
+ *
+ * - Each of these methods executes a single command in a process or subshell,
+ *   accepts a string for input to $stdin,
+ *   and returns string output from $stdout, $stderr, or both:
+ *
+ *   - Open3.capture2: Executes the command;
+ *     returns the string from $stdout.
+ *   - Open3.capture2e: Executes the command;
+ *     returns the string from merged $stdout and $stderr.
+ *   - Open3.capture3: Executes the command;
+ *     returns strings from $stdout and $stderr.
+ *
+ * - Each of these methods executes a single command in a process or subshell,
+ *   and returns pipes for $stdin, $stdout, and/or $stderr:
+ *
+ *   - Open3.popen2: Executes the command;
+ *     returns pipes for $stdin and $stdout.
+ *   - Open3.popen2e: Executes the command;
+ *     returns pipes for $stdin and merged $stdout and $stderr.
+ *   - Open3.popen3: Executes the command;
+ *     returns pipes for $stdin, $stdout, and $stderr.
+ *
+ * - Each of these methods executes one or more commands in processes and/or subshells,
+ *   returns pipes for the first $stdin, the last $stdout, or both:
+ *
+ *   - Open3.pipeline_r: Returns a pipe for the last $stdout.
+ *   - Open3.pipeline_rw: Returns pipes for the first $stdin and the last $stdout.
+ *   - Open3.pipeline_w: Returns a pipe for the first $stdin.
+ *   - Open3.pipeline_start: Does not wait for processes to complete.
+ *   - Open3.pipeline: Waits for processes to complete.
+ *
+ * - Process.spawn, Kernel#spawn: Executes the command;
+ *   returns the new pid without waiting for completion.
+ * - Process.exec: Replaces the current process by executing the command.
+ *
+ * In addition:
+ *
+ * - Kernel#system executes a given command-line (string) in a subshell;
+ *   returns +true+, +false+, or +nil+.
+ *
+ * Each of the methods above accepts:
+ *
+ * - An optional hash of environment variable names and values;
+ *   see {Exection Environment}[rdoc-ref:Process@Execution+Environment].
+ * - A required string argument that is a +command_line+ or +exe_path+;
+ *   see {Argument command_line or exe_path}[rdoc-ref:Process@Argument+command_line+or+exe_path].
+ * - An optional hash of execution options;
+ *   see {Execution Options}[rdoc-ref:Process@Execution+Options].
```

Comment:
I think this kind of detail should belong in the documentation for each respective method, rather than here where it's just an overview.

Reply:
@BurdetteLamar can you address this comment too?

Reply:
Removed here.  I think the three methods already have the details, so no changes there.